examples/model-score/app.py

```diff
@@ -0,0 +1,305 @@
+from shinywidgets import output_widget
+
+import shiny.experimental as x
+from shiny import App, Inputs, Outputs, Session, reactive, render, ui
+
+THRESHOLD_MID = 0.85
+THRESHOLD_MID_COLOR = "rgb(0, 137, 26)"
+THRESHOLD_LOW = 0.5
+THRESHOLD_LOW_COLOR = "rgb(193, 0, 0)"
+
+# Start a background thread that writes fake data to the SQLite database every second
+scoredata.begin()
+
+con = sqlite3.connect(scoredata.SQLITE_DB_URI, uri=True)
+
+
+def last_modified(con):
+    """
+    Fast-executing call to get the timestamp of the most recent row in the database.
+    We will poll against this in absence of a way to receive a push notification when
+    our SQLite database changes.
+    """
+    return con.execute("select max(timestamp) from auc_scores").fetchone()[0]
+
+
+@reactive.poll(lambda: last_modified(con))
+def df():
+    """
+    @reactive.poll calls a cheap query (`last_modified()`) every 1 second to check if
+    the expensive query (`df()`) should be run and downstream calculations should be
+    updated.
+
+    By declaring this reactive object at the top-level of the script instead of in the
+    server function, all sessions are sharing the same object, so the expensive query is
+    only run once no matter how many users are connected.
+    """
+    tbl = pd.read_sql(
+        "select * from auc_scores order by timestamp desc, model desc limit ?",
+        con,
+        params=[150],
+    )
+    # Convert timestamp to datetime object, which SQLite doesn't support natively
+    tbl["timestamp"] = pd.to_datetime(tbl["timestamp"], utc=True)
+    # Create a short label for readability
+    tbl["time"] = tbl["timestamp"].dt.strftime("%H:%M:%S")
+    # Reverse order of rows
+    tbl = tbl.iloc[::-1]
+
+    return tbl
+
+
+def read_time_period(from_time, to_time):
+    tbl = pd.read_sql(
+        "select * from auc_scores where timestamp between ? and ? order by timestamp, model",
+        con,
+        params=[from_time, to_time],
+    )
+    # Treat timestamp as a continuous variable
+    tbl["timestamp"] = pd.to_datetime(tbl["timestamp"], utc=True)
+    tbl["time"] = tbl["timestamp"].dt.strftime("%H:%M:%S")
+
+    return tbl
+
+
+model_colors = {
+    "model_1": "#7fc97f",
+    "model_2": "#beaed4",
+    "model_3": "#fdc086",
+    "model_4": "#ffff99",
+}
+model_names = list(model_colors.keys())
+
+
+def app_ui(req):
+    end_time = datetime.now(timezone.utc)
+    start_time = end_time - timedelta(minutes=1)
+
+    return x.ui.page_sidebar(
+        x.ui.sidebar(
+            ui.input_checkbox_group(
+                "models", "Models", model_names, selected=model_names
+            ),
+            ui.input_radio_buttons(
+                "timeframe",
+                "Timeframe",
+                ["Latest", "Specific timeframe"],
+                selected="Latest",
+            ),
+            ui.panel_conditional(
+                "input.timeframe === 'Latest'",
+                ui.input_selectize(
```

Comment:
I'm not sure that people will view this app long enough to get value out of the longer time intervals, would it be better to make this a numeric input with 5s steps?

Reply:
I didn't think people would actually sit there and wait, but I did think it was representative of what people might do in real life. I can put some other intervals in there though.